plugin.video.ampya/default.py

```diff
@@ -0,0 +1,341 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import sys
+import urlparse
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import xbmc
+import xbmcvfs
+import urllib, urllib2, socket, cookielib, re, os, shutil,json
+import time
+import datetime
+
+# Setting Variablen Des Plugins
+global debuging
+base_url = sys.argv[0]
+addon_handle = int(sys.argv[1])
+
+args = urlparse.parse_qs(sys.argv[2][1:])
+addon = xbmcaddon.Addon()
+# Lade Sprach Variablen
+translation = addon.getLocalizedString
+defaultBackground = ""
+defaultThumb = ""
+cliplist=[]
+filelist=[]
+profile    = xbmc.translatePath( addon.getAddonInfo('profile') ).decode("utf-8")
+temp       = xbmc.translatePath( os.path.join( profile, 'temp', '') ).decode("utf-8")
+#Directory für Token Anlegen
+if not xbmcvfs.exists(temp):       
+       xbmcvfs.mkdirs(temp)
+       
+
+icon = xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('path')+'/icon.png').decode('utf-8')
+useThumbAsFanart=addon.getSetting("useThumbAsFanart") == "true"
+
+
+
+def debug(content):
+    log(content, xbmc.LOGDEBUG)
+    
+def notice(content):
+    log(content, xbmc.LOGNOTICE)
+
+def log(msg, level=xbmc.LOGNOTICE):
+    addon = xbmcaddon.Addon()
+    addonID = addon.getAddonInfo('id')
+    xbmc.log('%s: %s' % (addonID, msg), level) 
+
+def parameters_string_to_dict(parameters):
+  paramDict = {}
+  if parameters:
+    paramPairs = parameters[1:].split("&")
+    for paramsPair in paramPairs:
+      paramSplits = paramsPair.split('=')
+      if (len(paramSplits)) == 2:
+        paramDict[paramSplits[0]] = paramSplits[1]
+  return paramDict
+  
+    
+def addDir(name, url, mode, iconimage, desc=""):
+  u = sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)
+  ok = True
+  liz = xbmcgui.ListItem(name, iconImage=icon, thumbnailImage=iconimage)
+  liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc})
+  if useThumbAsFanart:
+    if not iconimage or iconimage==icon or iconimage==defaultThumb:
+      iconimage = defaultBackground
+    liz.setProperty("fanart_image", iconimage)
+  else:
+    liz.setProperty("fanart_image", defaultBackground)
+  ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=True)
+  return ok
+  
+def addLink(name, url, mode, iconimage, duration="", desc="",artist_id="",genre="",shortname="",production_year=0,zeit=0,liedid=0):  
+  cd=addon.getSetting("password")  
+  u = sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)
+  ok = True
+  liz = xbmcgui.ListItem(name, iconImage=defaultThumb, thumbnailImage=iconimage)
+  liz.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Genre": genre,"Sorttitle":shortname,"Dateadded":zeit,"year":production_year })
+  liz.setProperty('IsPlayable', 'true')
+  liz.addStreamInfo('video', { 'duration' : duration })
+  liz.setProperty("fanart_image", iconimage)
```

Comment:
WIll in in the next Problem, just waiting how to solve the Loop, so i can make a new PR